latch/types/metadata.py

```diff
@@ -456,6 +476,108 @@ class SnakemakeFileMetadata:
     """
 
 
+@dataclass
+class NextflowParameter(Generic[T], LatchParameter):
+    type: Optional[Type[T]] = None
+    """
+    The python type of the parameter.
+    """
+    default: Optional[T] = None
+    """
+    Default value of the parameter
+    """
+
+    samplesheet_type: Literal["csv", "tsv", None] = None
+    """
+    The type of samplesheet to construct from the input parameter.
+
+    Only used if the provided parameter is a samplesheet (samplesheet=True)
+    """
+    samplesheet_constructor: Optional[Callable[[T], Path]] = None
+    """
+    A custom samplesheet constructor.
+
+    Should return the path of the constructed samplesheet. If samplesheet_type is also specified, this takes precedence.
+    Only used if the provided parameter is a samplesheet (samplesheet=True)
+    """
+
+    def __post_init__(self):
+        if not self.samplesheet or self.samplesheet_constructor is not None:
+            return
+
+        t = self.type
+        if get_origin(t) is not list or not is_dataclass(get_args(t)[0]):
+            click.secho(
+                dedent("""\
+                Samplesheets must be a list of dataclasses.
+                """),
```

Comment:
lol ik this is my diff but still

```suggestion
                "Samplesheets must be a list of dataclasses.",
```